docs/developer.md

```diff
@@ -0,0 +1,70 @@
+# Developer Guidelines
+
+## How to contribute
+
+Contributions to `scope` are made through [GitHub Pull Requests](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests), a set of proposed commits (or patches).
+
+To prepare, you should:
+
+- Create your own fork the [scope repository](https://github.com/ZwickyTransientFacility/scope) by clicking the "fork" button.
+
+- [Set up SSH authentication with GitHub](https://help.github.com/en/github/authenticating-to-github/connecting-to-github-with-ssh).
+
+- Clone (download) your copy of the repository, and set up a remote called `upstream` that points to the main Fritz repository.
+
+  ```sh
+  git clone git@github.com:<yourname>/scope
+  git remote add upstream git@github.com:ZwickyTransientFacility/scope
+  ```
+
+Then, for each feature you wish to contribute, create a pull request:
+
+1. Download the latest version of `scope`, and create a new branch for your work.
+
+   Here, let's say we want to contribute some documentation fixes; we'll call our branch `rewrite-contributor-guide`.
+
+   ```sh
+   git checkout main
+   git pull upstream main
+   git checkout -b rewrite-contributor-guide
+   ```
+
+2. Make modifications to `scope` and commit your changes using `git add` and `git commit`.
+Each commit message should consist of a summary line and a longer description, e.g.:
+
+   ```text
+   Rewrite the contributor guide
+
+   While reading through the contributor guide, I noticed several places
+   in which instructions were out of order. I therefore reorganized all
+   sections to follow logically, and fixed several grammar mistakes along
+   the way.
+   ```
+
+3. When ready, push your branch to GitHub:
+
+   ```sh
+   git push origin rewrite-contributor-guide
+   ```
+
+   Once the branch is uploaded, GitHub should print a URL for turning your branch into a pull request.  Open that URL in your browser, write an informative title and description for your pull request, and submit it.
+
+4. The team will now review your contribution, and suggest changes. *To simplify review, please limit pull requests to one logical set of changes.* To incorporate changes recommended by the reviewers, commit edits to your branch, and push to the branch again (there is no need to re-create the pull request, it will automatically track modifications to your branch).
+
```

Comment:
Maybe here add a few lines showing how to rebase as is required at times if there are changes on master affecting your branch?

git fetch upstream
git rebase upstream/master

Reply:
Of course, it's mostly written by no other than the almighty @stefanv himself!
Added some bla-bla on merging vs rebasing.

Reply:
Our policy nowadays is to squash the PR merge, in which case it does not matter if developers merge master into their branch (as many times as they want to).  This simplifies the process quite a bit.

Reply:
Ah, right, good point! Will fix the doc.